tonic/src/body.rs

```diff
@@ -11,7 +11,14 @@ where
     B: http_body::Body<Data = bytes::Bytes> + Send + 'static,
     B::Error: Into<crate::Error>,
 {
-    body.map_err(crate::Status::map_error).boxed_unsync()
+    let mut body = Some(body);
+    if let Some(body) = <dyn std::any::Any>::downcast_mut::<Option<BoxBody>>(&mut body) {
```

Comment:
Is there evidence of this being an actual problem in the wild? Why do we need the `Option`-wrapping for this?

Reply:
> Is there evidence of this being an actual problem in the wild?

The motivation is to reduce the effect of the changes like #2001. I have not measured whether duplicate boxes actually affect performance.

> Why do we need the Option-wrapping for this?

This is because of Rust's ownership system. It appears that this is necessary to allow the argument to be reused when the downcasting of the body fails.